UpKeep/Controllers/InventoryItemController.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using UpKeep.Models;
+using UpKeep.Services.Interfaces;
+
+namespace UpKeep.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]/[action]")]
+    public class InventoryItemController : ControllerBase
+    {
+        private readonly ILogger<InventoryItemController> _logger;
+        private readonly IInventoryItemService _inventoryItemService;
+
+        public InventoryItemController(ILogger<InventoryItemController> logger, IInventoryItemService inventoryItemService)
+        {
+            _logger = logger;
+            _inventoryItemService = inventoryItemService;
+        }
+
+        [HttpGet, ActionName("GetInventoryItems")]
+        [ProducesResponseType(typeof(IEnumerable<InventoryItemModel>), StatusCodes.Status200OK)]
+        public async Task<IActionResult> GetInventoryItems()
+        {
+            return Ok(await _inventoryItemService.GetInventoryItems());
+        }
+
+        [HttpGet, ActionName("GetInventoryItemById")]
+        [ProducesResponseType(typeof(InventoryItemModel), StatusCodes.Status200OK)]
+        public async Task<IActionResult> GetInventoryItemById(int id)
+        {
+            return Ok(await _inventoryItemService.GetInventoryItemModelById(id));
+        }
+
+        [HttpPost, ActionName("UpdateInventoryItem")]
+        [ProducesResponseType(typeof(int), StatusCodes.Status200OK)]
+        public async Task<IActionResult> UpdateInventoryItem(InventoryItemModel inventoryItemModel)
+        {
+            return Ok(await _inventoryItemService.UpdateInventoryItem(inventoryItemModel));
+        }
+
+        [HttpPost, ActionName("AddInventoryItem")]
+        [ProducesResponseType(typeof(int), StatusCodes.Status200OK)]
+        public async Task<IActionResult> AddInventoryItem(InventoryItemModel inventoryItemModel)
+        {
+            return Ok(await _inventoryItemService.AddInventoryItem(inventoryItemModel));
+        }
+
+        [HttpPost, ActionName("DeleteInventoryItem")]
+        [ProducesResponseType(typeof(int), StatusCodes.Status200OK)]
+        public async Task<IActionResult> DeleteInventoryItem(InventoryItemModel inventoryItemModel)
+        {
+            return Ok(await _inventoryItemService.DeleteInventoryItem(inventoryItemModel));
+        }
+    }
+}
```

Comment:
good start on the controllers getting the actions posted is the next step adding to these with like a fetch by ID and a fetchall() or a fetch by room - which might be on another page.